test/e2e/gameserverallocation_test.go

```diff
@@ -126,10 +126,16 @@ func TestGameServerAllocationMetaDataPatch(t *testing.T) {
 			},
 		}}
 
-	gsa, err = framework.AgonesClient.AllocationV1alpha1().GameServerAllocations(defaultNs).Create(gsa.DeepCopy())
-	if assert.Nil(t, err) {
-		assert.Equal(t, v1alpha1.GameServerAllocationAllocated, gsa.Status.State)
-	}
+	err = wait.PollImmediate(time.Second, 30*time.Second, func() (bool, error) {
+		gsa, err = framework.AgonesClient.AllocationV1alpha1().GameServerAllocations(defaultNs).Create(gsa.DeepCopy())
```

Comment:
i believe this will cause panic if "err" is ever returned, because "gsa" will be set to nil and next iteration will use nill.DeepCopy()

Reply:
Good catch. Fixed!